presto-main/src/main/java/com/facebook/presto/cost/HistoricalStatisticsProvider.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.cost;
+
+import com.facebook.presto.spi.statistics.HistoricalPlanStatistics;
+import com.facebook.presto.spi.statistics.HistoricalStatisticsStorageService;
+import com.facebook.presto.spi.statistics.PlanStatistics;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+
+public class HistoricalStatisticsProvider
```

Comment:
A natural question would be: what is the relationship between `HistoricalStatisticsProvider` and `ExternalPlanStatisticsProvider`? Can one be the implementation of the other?